tests/dragonfly/replication_test.py

```diff
@@ -444,7 +444,8 @@ async def replicate():
             await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
             return True
         except redis.exceptions.ResponseError as e:
-            assert e.args[0] == "replication cancelled"
+            err = e.args[0]
+            assert err == "replication cancelled" or err == "Can not execute during LOADING"
```

Comment:
why did you add this check - Can not execute during LOADING
we should block REPLICAOF command when loading only if we load from snapshot, not if we run another replicaof command and in loading state due to full sync

Reply:
The condition here https://github.com/dragonflydb/dragonfly/pull/2338/files#diff-04447876005188a028fe0e46ee54f437b03f9d2dfb8b456be2fa6fa539a56499R1946 will trigger even if we are not loading from snapshot. Imagine two REPLICAOF commands issued one after the other. 

First one is a little faster, `switches the state to loading` and it just executed the statement:
 `2063   auto new_replica = make_shared<Replica>(string(host), port, &service_, master_id());` (just before it `SetMasterFlagOnAllThreads(false);`). Second, `REPLICAOF` command arrives at the branch: 
 
 `  2016   if (ServerState::tlocal()->is_master && service_.GetGlobalState() == GlobalState::LOADING) {  ` 
 
 And now it gets rejected. 
 
 I guess this is not true: 
 > we should block REPLICAOF command when loading only if we load from snapshot,
 for this particular case. 
 
 We do not have a global state specific to loading snapshots...

Reply:
in the senario you are descibing above which we see the loading state because we started replication but the the set master to false did not happen yet should not occur as we are under the replicaof_mu_ so when executing replicaof after another replicaof we must see both loading state and is_master=false